dbms/src/Processors/Pipe.cpp

```diff
@@ -54,6 +54,10 @@ Pipe::Pipe(ProcessorPtr source)
 {
     checkSource(*source);
     output_port = &source->getOutputs().front();
+
+    if (source->getOutputs().size() > 1)
+        totals = &source->getOutputs().back();
```

Comment:
Totals are always the last port?

Reply:
Yes for now.
I plan to improve an interface of `Pipe` and `ISource` to better support totals.